### ISSUE TITLE
False positive from RS1024 (Compare symbols correctly) when initializing a HashSet<ITypeSymbol>

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn-analyzers#5246]
### Analyzer

**Diagnostic ID**: RS1024: `Compare symbols correctly`

### Analyzer source

Package: Microsoft.CodeAnalysis.CSharp.Workspaces
**Version**: 3.10.0

### Describe the bug

This line produces the RS1024 diagnostic, even though we *do* specify to use the `SymbolEqualityComparer`:

```cs
private readonly HashSet<ITypeSymbol> alreadyCollected = new(SymbolEqualityComparer.Default);
```

**Comment by Youssef1313** [comment]
This was fixed in https://github.com/dotnet/roslyn-analyzers/pull/4571. I'm not sure which version (if any) has the fix.

**Comment by Evangelink** [comment]
@AArnott could you confirm the issue is fixed?

**Comment by AArnott** [comment]
It still repros with 4.0.0-2.final (the latest on nuget.org.

**Comment by Evangelink** [comment]
I have managed to see the same issue on roslyn-analyzer itself that's why I was asking this question.

**Comment by ryzngard** [comment]
@mavasani is it just that we haven't released this fix yet?

**Comment by zivarah** [comment]
I would appreciate confirmation on whether this has been fixed/released yet and, if so, what version contains the fix.

**Comment by 333fred** [comment]
@mavasani I'm still seeing this. Have we released a version this this fix?

**Comment by Youssef1313** [comment]
@333fred Are you using Microsoft.CodeAnalysis.Analyzers 3.3.3?

**Comment by Youssef1313** [comment]
> It still repros with 4.0.0-2.final (the latest on nuget.org.)

@AArnott The analyzer comes from `Microsoft.CodeAnalysis.Analyzers` and the latest version is 3.3.3.

The Workspaces package you're referring to has the Analyzers package as a dependency, and it unfortunately doesn't reference the latest. So you have to explicitly install `Microsoft.CodeAnalysis.Analyzers` 3.3.3.

I opened https://github.com/dotnet/roslyn/pull/58154 on roslyn side to reference the latest package.

**Comment by 333fred** [comment]
> @333fred Are you using Microsoft.CodeAnalysis.Analyzers 3.3.3?

I was not explicitly referencing the package, so I was using whatever version the O# roslyn dependency version pulls in (I think 3.3.2?).

**Comment by Youssef1313** [comment]
@333fred If it was pulled from one of the other `Microsoft.CodeAnalysis.*` packages, then yes it would be 3.3.2. dotnet/roslyn#58154 should fix that.

**Comment by Youssef1313** [comment]
@333fred I've also updating PostReleaseActivity.md in https://github.com/dotnet/roslyn-analyzers/pull/5757 to note that new versions should be used by dotnet/roslyn to avoid this kind of issues. I assume @JoeRobich and @mavasani follow these instructions when releasing a new version?

**Comment by Youssef1313** [comment]
I believe this can be closed.
